config.go

```diff
@@ -138,8 +143,76 @@ type HumanConfig struct {
 	PingType flags.StringValue `mapstructure:"ping_type"`
 
 	DisableCoordinateUpdates flags.BoolValue `mapstructure:"disable_coordinate_updates"`
+
+	MaxFailingChecks UintValue `mapstructure:"max_failing_checks"`
+	MinPassingChecks UintValue `mapstructure:"min_passing_checks"`
+}
+
+// UintValue provides a flag value that's aware if it has been set.
+type UintValue struct {
+	v *uint
+}
+
+// Merge will overlay this value if it has been set.
+func (u *UintValue) Merge(onto *uint) {
+	if u.v != nil {
+		*onto = *(u.v)
+	}
+}
+
+// Set implements the flag.Value interface.
+func (u *UintValue) Set(v string) error {
+	if u.v == nil {
+		u.v = new(uint)
+	}
+	parsed, err := strconv.ParseUint(v, 0, 64)
+	*(u.v) = (uint)(parsed)
+	return err
+}
+
+// String implements the flag.Value interface.
+func (u *UintValue) String() string {
+	var current uint
+	if u.v != nil {
+		current = *(u.v)
+	}
+	return fmt.Sprintf("%v", current)
+}
+
+func intToUintValueFunc() mapstructure.DecodeHookFunc {
```

Comment:
Most of the changes related to conversion from `int` to `uint` can be avoided if just `int` is used. Again, if we refer to the Consul's codebase, health checking math is performed using the `int` type variables, e.g. [consul/agent/check/check.go:810-813](https://github.com/hashicorp/consul/blob/3570ce6566d207787d78fa91f99e9847222ad8a8/agent/checks/check.go#L810-L813).

Reply:
The thing is that we need a type, `UintValue`, that is aware if it has been set or not for the Merge() afterwards. And we need a function to convert what we get, `int`, into that type. Since there was no function to convert to the existing `UintValue` from int, I needed to copy some code from Consul here, and implement `intToUintValueFunc()`. Using an hypothetical `IntValue` instead of `UintValue` would not save us any work, and uint makes more sense as a type for these values.

Reply:
Well, by using an `int` the model would allow negative values for `SuccessBeforePassing` and `FailuresBeforeCritical`. What would this mean? These would be errors, right? I always prefer to make illegal states unrepresentable.

Or do you have some valid behavior in mind for negative values for these configuration settings?

Reply:
Thanks for sharing your thinking. I see what you're saying - `uint` more accurately represents the number range we would want to cover with a counter than `int`

In application, I prefer `int` for a number of reasons. Typically the use of `uint`s is heavily associated with bitmasks and such, and not so much counters. `uint`s are often not recommended since they have the rollover at 0 - this can surprise people especially since they are not commonly used. `int`s are often used to represent states even though they don't have negative values - e.g. array lengths are represented by ints. The added complexity of using `uint` doesn't seem to outweigh the benefit of not representing negative values to me. For example, the `decrementCounter()` is needed regardless of `uint` or `int`